pystiche_papers/ulyanov_et_al_2016/_data.py

```diff
@@ -186,71 +187,71 @@ def images() -> DownloadableImageCollection:
     return DownloadableImageCollection({**content_images, **style_images})
 
 
+class SkipSmallIterableDataset(torch.utils.data.IterableDataset):
+    def __init__(
+        self,
+        dataset: torch.utils.data.Dataset,
+        min_size: int,
+        num_samples: int,
+        transform: transforms.Transform,
+    ):
+        self.dataset = dataset
+        self.min_size = min_size
+        self.num_samples = num_samples
+        self.transform = transform
+        self.data_samples = itertools.cycle(self.dataset)
```

Comment:
Why do this in the constructor? This means the dataset is not "reset" if whenever we call `iter` on it. For example:

```py
import itertools


class Dataset:
    def __init__(self):
        self.data_samples = itertools.cycle([1, 2, 3])

    def __iter__(self):
        yield from self.data_samples


dataset = Dataset()

for epoch in range(2):
    for idx, sample in enumerate(dataset):
        print(sample)
        if idx == 3:
            break
```

```
1
2
3
1
2
3
1
2
```

Compare that to

```py
class Dataset:
    def __init__(self):
        self.dataset = [1, 2, 3]

    def __iter__(self):
        yield from itertools.cycle(self.dataset)
```

```
1
2
3
1
1
2
3
1
```

Or is this behavior intended here?

Reply:
Yes, this is intended here so that the dataset is not reset after each epoch.